python/python/bystro/ancestry/train.py

```diff
@@ -308,6 +313,64 @@ def _calc_fst(variant_counts: pd.Series, samples: pd.DataFrame) -> float:
     return (p * (1 - p) - total) / (p * (1 - p))
 
 
+def load_1kgp_vcf(TKGP_VCF):
+    """This is a temporary placeholder method to load in 1kgp vcf 
+    filtered down to the same variants as the gnomad loadings for testing
+    using plink2"""
+    headervcf=pd.read_csv(TKGP_VCF,delimiter='\t',comment='#')
+    dosagevcf = headervcf.replace('0|0', 0)
+    dosagevcf = dosagevcf.replace('0|1', 1)
+    dosagevcf = dosagevcf.replace('1|0', 1)
+    dosagevcf = dosagevcf.replace('1|1', 2)
+    dosagevcf=dosagevcf.rename(columns={"#CHROM": "Chromosome", "POS": "Position"}, errors="raise")
+    return dosagevcf
+    
+
+def load_pca_loadings(loadings, dosage_vcf):
+    """This loads in the gnomad PCs, reformats, and filters them to match
+    the format and variant list of the 1kgp vcf"""
+    loadings[['Chromosome', 'Position']] = loadings['locus'].str.split(':', expand=True)
+    #Match variant format with vcf
+    loadings['variant']=loadings.apply(lambda x: ':'.join([x['locus'][3:],x['alleles'][2],x['alleles'][6]]),axis=1)
+    #Remove variants that are missing from IGSR version of 1kgp
+    missing_variants = set(loadings.variant) - set(dosagevcf.ID)
+    loadings=loadings[~loadings['variant'].isin(missing_variants)]
+    #Remove brackets
+    loadings['loadings'] = loadings['loadings'].str[1:]
+    loadings['loadings'] = loadings['loadings'].str[:-1]
+    #Split PCs and join back
+    gnomadPCs = loadings['loadings'].str.split(',', expand=True)
+    loadings=loadings.join(gnomadPCs)
+    loadings=loadings.reset_index()
+    return loadings
+
+
+def apply_pca_transform(loadings, dosagevcf):
+    """Transforms vcf with genotypes in dosage format with PCs loadings from gnomad PCA"""
+    #Both files need to be np arrays for pca transformation
+    pc_loadings = loadings.iloc[:, 8:28]
+    genos = dosagevcf.iloc[:, 9:]
+    scaler = StandardScaler()
+    TGP_scaled = scaler.fit_transform(genos).T
+    TGP_scaled = np.array(TGP_scaled)
+    pc_loadings_array = pc_loadings.values.astype(float)
+    #Ensure the number of features matches the original dataset
+    if TGP_scaled.shape[1] != pc_loadings.shape[0]:
```

Comment:
this is a good check but we should also guarantee that the order of variants in pc_loadings agrees with the order of variants in the TGP file.  (Otherwise we'd get a silent failure here.)